### ISSUE TITLE
Paper for Minecraft 1.13 cannot load chunks created by Minecraft 1.12.2

### ISSUE THREAD

**Issue by rutgerkok** [PaperMC/Paper#1348]
### What behaviour is expected:
When loading a world created in Minecraft 1.12.2, the chunks should be converted to the latest save format.

### What behaviour is observed:
Chunks from Minecraft 1.12.2 are discarded; entirely new terrain is generated. Player positions are still preserved, however.

### Steps/models to reproduce:
1. Create a world in Minecraft 1.12.2. (Both a four year old server world and a world freshly generated by singleplayer Minecraft 1.12.2 have this issue. I have uploaded a small example world here: https://www.dropbox.com/s/2587wqosedenclm/Minecraft%201.12%20world.zip?dl=0 . Near the spawn point, the letters HI have been built using orange wool. There is also a small sandstone road.)
2. Open the world in a recent Paper 1.13 build. Notice that the terrain is regenerated.

### Plugin list:
* I was able to reproduce this issue on a server without plugins, using a world generated in vanilla Minecraft.

### Paper build number:
This server is running Paper version git-Paper-150 (MC: 1.13) (Implementing API version 1.13-R0.1-SNAPSHOT)

### Anything else:
Build  git-Paper-114 (MC: 1.13) does not have this issue. Build git-Paper-115 (MC: 1.13) introduced the issue. Build git-Paper-150 (MC: 1.13) still shows this issue. Therefore, I believe this issue is caused by commit 6666adc92ccd953fea83a5369fae1b8280c1200d .

**Comment by BillyGalbreath** [comment]
Uh.. it's common knowledge that converting legacy world's to 1.13 is broken. It's been discussed and mentioned everywhere, including upstream in Soigot.

You will need to wait for 1.13.1 if you want to bring a legacy world into 1.13

**Comment by zachbr** [comment]
Broken as in potential data loss when running conversions and fixers. Not regenerating all the chunks entirely.

**Comment by zachbr** [comment]
I've seen this happen in passing exactly one time.
I'm going to move this to `bug` and try and look at it tonight.

This is something that we really need to double-check and be sure about with 1.13.1 coming, and presumably lots of servers wanting to do world conversions.

**Comment by rutgerkok** [comment]
I can consistently reproduce this issue using the linked world. I'm using the latest Paper build (160). In this image we can only see newly generated terrain. I'm using the world I linked in the opening post.

![Paper](https://user-images.githubusercontent.com/1462188/44539675-18f22e80-a705-11e8-8056-07b894ef13c0.png)

(I'm standing near the spawn point in both images, at `xyz: -221 67 249`.) I should note that Spigot does not have this bug.  I just tested the latest version, which is `git-Spigot-fe3ab0d-3418f8d`:

![Spigot](https://user-images.githubusercontent.com/1462188/44539197-d11ed780-a703-11e8-92f2-a893b1105521.png)

To me it seems that Minecraft converts 1.12-chunks to 1.13-ProtoChunks, probably so that the block lighting is forcibly recalculated. (This recalculation is necessary because water now lets more light through.) However, as of commit https://github.com/PaperMC/Paper/commit/6666adc92ccd953fea83a5369fae1b8280c1200d Paper simply drops those ProtoChunks, forcing them to regenerate.

At least, this theory explains why I can only observe this bug in Paper build 115 and newer, and not in Spigot and Paper build 114.
